vignettes/checks/plausibleAfterBirth.Rmd

```diff
@@ -0,0 +1,94 @@
+
+- *Numerator*: The number of records with a non-null date value that happen prior to birth.
+- *Denominator*: The total number of records in the table with a non-null date value.
+- *Related CDM Convention(s)*: -Not linked to a convention-
+- *CDM Fields/Tables*: By default, this check runs on all date and datetime fields
+- *Default Threshold Value*: 1
+
+## User Guidance
+There might be valid reasons why a record has a date value that occurs prior to birth. For example, prenatal observations might be captured or procedures on the mother might be added to the file of the child. Therefore, some failing records are expected and the default threshold of 1% accounts for that.
+
+However, if more records violate this check, there might be an issue with incorrect birthdates or events with a default date. It is recommended to investigate the records that fail this check to determine the cause of the error and set proper dates. If it is impossible to fix, then implement one of these:
+
+ - Aggressive: Remove all patients who have at least one record before birth (if the birthdate of this patient is unreliable)
+ - Less aggressive: Remove all rows that happen before birth. Probably this should be chosen as a conventional approach for data clean up (if the event dates are unreliable)
+ - Conservative: set event date to birth date.
+
+Make sure to clearly document the choices in your ETL specification.
+
+
+### Violated rows query
+You may also use the “violated rows” SQL query to inspect the violating rows and help diagnose the potential root cause of the issue:
+
+```sql
+SELECT p.birth_datetime, cdmTable.*
+FROM @cdmDatabaseSchema.@cdmTableName cdmTable
+JOIN @cdmDatabaseSchema.person p ON cdmTable.person_id = p.person_id
+WHERE cdmTable.@cdmFieldName < p.birth_datetime, 
+```
+or, when birth_datetime is missing:
+```sql
+SELECT p.birth_datetime, cdmTable.*
+FROM @cdmDatabaseSchema.@cdmTableName cdmTable
+JOIN @cdmDatabaseSchema.person p ON cdmTable.person_id = p.person_id
+WHERE cdmTable.@cdmFieldName < CAST(CONCAT(
+        p.year_of_birth,
+        COALESCE(
+            RIGHT('0' + CAST(p.month_of_birth AS VARCHAR), 2),
+            '01'
+        ),
+        COALESCE(
+            RIGHT('0' + CAST(p.day_of_birth AS VARCHAR), 2),
+            '01'
+        )
+    ) AS DATE)
+```
+
+Also, the length of the time interval between these dates might give you a hint of why the problem appears.
+```sql
+select date_difference, count(*)
+from (
+    select daydiff(
+        day, 
+        @cdmFieldName, 
+        COALESCE(
+            CAST(p.birth_datetime AS DATE),
+            CAST(CONCAT(p.year_of_birth,'-01-01') AS DATE))
+        ) as date_difference
+    from @cdmTableName ct
+    join person p on ct.person_id = p.person_id 
+) cte
+where date_difference < 0
```

Comment:
should be > 0 ?  i think datediff does second date - first date

Reply:
Indeed! Thanks